File: Java/Easy/0069-sqrt(x).java
class Sqrt {
    public int mySqrt(int x) {
        int left = 0;
        int right = x;
        int res = 0;
        while (left <= right) {
            int middle = left + ((right - left) / 2);
            if ((long) middle * middle > x) {
                right = middle - 1;
            } else if ((long) middle * middle < x) {
                left = middle + 1;
                res = middle;
            } else {
                return middle;
            }
        }
        return res;
    }
}

File: JavaScript/Easy/0069-sqrt(x).js
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function(x) {
    let left = 0;
    let right = x;
    let res = 0;
    while (left <= right) {
            let middle = Math.floor(left + ((right - left) / 2));
            if (middle * middle > x) {
                right = middle - 1;
            } else if (middle * middle < x) {
                left = middle + 1;
                res = middle;
            } else {
                return middle;
            }
        }
        return res;
};